client/landing/stepper/declarative-flow/internals/steps-repository/design-choices/index.tsx

```diff
@@ -88,7 +87,7 @@ const DesignChoicesStep: Step = ( { navigation, flow, stepName } ) => {
 								destination="pattern-assembler"
 								onSelect={ handleSubmit }
 							/>
-							{ isBigSkyEligible && recordBigSkyView() && (
+							{ ! isLoading && isEligible && recordBigSkyView() && (
```

Comment:
Thanks for adding the loading!

Note that `recordBigSkyView` will probably be called more than one time. If the component re-renders for any reason, the event will fire again, and there are virtually infinite reasons for the component to re-render. It should be called inside a `useEffect` with the boolean conditions.

```js
useEffect( () => {
	if ( ! isLoading && isEligible ) {
		recordTracksEvent( 'calypso_big_sky_view_choice', {
			flow,
			step: stepName,
		} );
	}
}, [ isEligible, isLoading, flow, stepName ] );
```